animation_url/src/routes/[slug]/+page.ts

```diff
@@ -31,19 +49,64 @@ export async function load({ url }) {
   });
   const httpUri = ipfsGatewayUri + metadata.image.slice(7);
 
-  // get pilot
-  let pilot;
-  const sessions = await tableland.read(
-    `SELECT end_time FROM ${deployment.pilotSessionsTable} WHERE rig_id = ${rigId} AND end_time is null;`,
-    { output: "objects" }
-  );
-  if (sessions && sessions.length > 0) {
-    pilot = trainer;
-  }
+  const pilot = await getPilot(rigId);
 
   return {
     imageUrl: httpUri,
     badges: [],
     pilot,
   };
 }
+
+const getPilot = async function (rigId, pilotAddress, pilotId): Promise<string> {
+  // check if rig is parked first
+  const notParked = await tableland.read(
+    `SELECT end_time FROM ${deployment.pilotSessionsTable} WHERE rig_id = ${rigId} AND end_time is null;`, {
+      output: "objects"
+    }
+  );
+
+  // parked
+  if (!notParked || notParked.length === 0) {
+    return "";
+  }
+
+  // get the 2 newest sessions so we can decide if the rig is training or flying
+  const sessionArr = await tableland.read(
+    `SELECT * FROM ${deployment.pilotSessionsTable} WHERE rig_id = ${rigId} ORDER BY start_time desc LIMIT 2;`, {
+      output: "objects"
+    }
+  );
+
+  // If the rig only has one session and end_time is null, it's in training
+  if (sessionArr.length === 1 && sessionArr[0].end_time === null) {
+    return trainer;
+  }
+
+  const currentSession = sessionArr && sessionArr[0];
+  // no session means nothing to show because the rig has never been trained
+  if (!currentSession) {
+    return "";
+  }
+
+  // if end_time is greater than or equal to start_time it's parked or never finished training
+  if (currentSession.end_time >= currentSession.start_time) {
+    return "";
+  }
+
+  // If we get here it means the rig is piloted. NOTE: this will only show eth mainnet Pilots
+  if (currentSession.pilot_contract && currentSession.pilot_id) {
+    const pilotToken = await zdk.token({
+      token: {
+        address: currentSession.pilot_contract, // "0x6c9343ca5c2ef3a35a83438344bb3cbe3c249f65",
+        tokenId: currentSession.pilot_id.toString(), // "903",
```

Comment:
You can uncomment the values here if you want to testout what the pilots look like during development